scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamMode.scala

```diff
@@ -74,3 +78,54 @@ object BeamSink extends Serializable {
         pc.asInstanceOf[PCollection[String]].apply(TextIO.write().to(path))
     }
 }
+
+class BeamFileIO[T](output: String) extends BeamSink[T] {
+  override def write(
+      pipeline: Pipeline,
+      config: Config,
+      pc: PCollection[_ <: T]
+  ): Unit = {
+    val pColT: PCollection[T] = BeamFunctions.widenPCollection(pc)
+
+    pColT.apply(
+      FileIO
+        .write()
+        .via(new CoderFileSink(pColT.getCoder))
+        .to(output)
+        .withNumShards(1)
+    )
+  }
+}
+
+class CoderFileSink[T](coder: Coder[T]) extends FileIO.Sink[T] {
+  private var outputStream: java.io.OutputStream = _
+
+  override def open(channel: WritableByteChannel): Unit =
+    outputStream = Channels.newOutputStream(channel)
+
+  override def write(element: T): Unit = coder.encode(element, outputStream)
+  override def flush(): Unit = outputStream.flush()
```

Comment:
is there no `close` in `FileIO.Sink`? Seems like we should have a way to close outputStream.

Reply:
It is handled by the underlying FileIO which uses FileIO.Sink